gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -827,19 +781,62 @@ func (vm *VMKeeper) QueryFile(ctx sdk.Context, filepath string) (res string, err
 		if memFile == nil {
 			return "", fmt.Errorf("file %q is not available", filepath) // TODO: XSS protection
 		}
+
 		return memFile.Body, nil
-	} else {
-		memPkg := store.GetMemPackage(dirpath)
-		if memPkg == nil {
-			return "", fmt.Errorf("package %q is not available", dirpath) // TODO: XSS protection
+	}
+
+	memPkg := store.GetMemPackage(dirpath)
+	if memPkg == nil {
+		return "", fmt.Errorf("package %q is not available", dirpath) // TODO: XSS protection
+	}
+
+	for i, memfile := range memPkg.Files {
+		if i > 0 {
+			res += "\n"
+		}
+		res += memfile.Name
+	}
+
+	return res, nil
+}
+
+func stringifyResultValues(m *gno.Machine, format Format, values []gno.TypedValue) string {
```

Comment:
implement or add a comment to add gasmeter here.

Reply:
As this method inherits the `gno.Machine`, I think it naturally handles gas here.
@piux2 can you confirm ?